libs/cardano-ledger-core/src/Cardano/Ledger/State/CertState.hs

```diff
@@ -181,32 +181,34 @@ instance NoThunks (Accounts era) => NoThunks (DState era)
 instance NFData (Accounts era) => NFData (DState era)
 
 instance (Era era, EncCBOR (Accounts era)) => EncCBOR (DState era) where
-  encCBOR (DState unified fgs gs ir) =
-    encodeListLen 4
-      <> encCBOR unified
-      <> encCBOR fgs
-      <> encCBOR gs
-      <> encCBOR ir
+  encCBOR dState@(DState _ _ _ _) =
```

Comment:
Why do you need the `@(DState _ _ _ _)` part?

Also, why not have the wildcard deconstruction in the param? I think we talked about this before in connection with encoders and decoders but I don't remember the reason now.